File: samplejava/src/main/java/com/example/chattutorialjava/AttachmentViewHolderImgur.java
package com.example.chattutorialjava;

import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.getstream.sdk.chat.adapter.AttachmentListItem;
import com.getstream.sdk.chat.adapter.MessageListItem.MessageItem;
import com.getstream.sdk.chat.adapter.viewholder.attachment.BaseAttachmentViewHolder;
import com.getstream.sdk.chat.utils.roundedImageView.PorterShapeImageView;
import com.getstream.sdk.chat.view.MessageListView.BubbleHelper;

import org.jetbrains.annotations.NotNull;

import coil.Coil;
import coil.request.ImageRequest;

public class AttachmentViewHolderImgur extends BaseAttachmentViewHolder {
    private final BubbleHelper bubbleHelper;
    private final MessageItem messageItem;
    private final PorterShapeImageView imageView;

    public AttachmentViewHolderImgur(
            ViewGroup parent,
            BubbleHelper bubbleHelper,
            MessageItem messageItem
    ) {
        super(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.list_item_attach_imgur, parent, false));
        this.bubbleHelper = bubbleHelper;
        this.messageItem = messageItem;
        imageView = itemView.findViewById(R.id.iv_media_thumb);
    }

    @Override
    public void bind(@NotNull AttachmentListItem attachmentListItem) {
        Drawable background = bubbleHelper.getDrawableForAttachment(
                messageItem.getMessage(),
                messageItem.isMine(),
                messageItem.getPositions(),
                attachmentListItem.getAttachment());
        imageView.setShape(getContext(), background);

        ImageRequest request = new ImageRequest.Builder(getContext())
                .data(attachmentListItem.getAttachment().getThumbUrl())
                .allowHardware(false)
                .target(imageView)
                .build();
        Coil.imageLoader(getContext()).enqueue(request);
    }
}

File: app/src/main/java/com/example/chattutorial/AttachmentViewHolderImgur.kt
package com.example.chattutorial

import android.view.LayoutInflater
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.getstream.sdk.chat.adapter.AttachmentListItem
import com.getstream.sdk.chat.adapter.MessageListItem
import com.getstream.sdk.chat.adapter.viewholder.attachment.BaseAttachmentViewHolder
import com.getstream.sdk.chat.utils.roundedImageView.PorterShapeImageView
import com.getstream.sdk.chat.view.MessageListView
import kotlinx.android.synthetic.main.list_item_attach_imgur.view.*

class AttachmentViewHolderImgur(
    parent: ViewGroup,
    private val bubbleHelper: MessageListView.BubbleHelper,
    private val messageItem: MessageListItem.MessageItem
) : BaseAttachmentViewHolder(
    LayoutInflater.from(parent.context)
        .inflate(R.layout.list_item_attach_imgur, parent, false)
) {

    private val ivMediaThumb: PorterShapeImageView = itemView.iv_media_thumb

    override fun bind(attachmentListItem: AttachmentListItem) {
        val background = bubbleHelper.getDrawableForAttachment(
            messageItem.message,
            messageItem.isMine,
            messageItem.positions,
            attachmentListItem.attachment
        )
        ivMediaThumb.setShape(context, background)

        Glide.with(context)
            .load(attachmentListItem.attachment.thumbUrl)
            .into(ivMediaThumb)
    }
}
